### ISSUE TITLE
CTRL-C should exit napari gracefully

### ISSUE THREAD

**Issue by pwinston** [napari/napari#1493]
## 🐛 Bug

After #1476 napari just prints `KeyboardInterrupt` when CTRL-C is pressed in the system terminal window that used to launch napari.  Prior to 1476 it exited with a crash, which got the job done but was not great. 

Ideally napari would exit gracefully when you hit CTRL-C in the system terminal window.

## To Reproduce

1. From a system terminal (e.g. Terminal program on mac)
2. Run "napari" or a script that uses `napari.gui_qt()`
3. Switch back to the terminal window and type CTRL-C

## Expected behavior

Napari exits gracefully.

## Environment

```
napari: not-installed
Platform: macOS-10.15.3-x86_64-i386-64bit
Python: 3.8.1 (default, Jan 8 2020, 16:15:59) [Clang 4.0.1 (tags/RELEASE_401/final)]
Qt: 5.14.2
PyQt5: 5.14.2
NumPy: 1.18.4
SciPy: 1.4.1
Dask: 2.17.2
VisPy: 0.6.5.dev111+g8387ea1a.d20200424

GL version: 2.1 ATI-3.5.5
MAX_TEXTURE_SIZE: 16384

Plugins:
- napari-plugin-engine: 0.1.6
- svg: 0.1.3
```

## Additional context

This is low priority since you can exit with the Quit command, or from the system terminal hit CTRL-Z and `kill %1` the app if necessary. However it seems like exiting gracefully is the right behavior long term.

I tried adding this to our new `ExceptionHandler` class:
```
        # Interpret CTRL-C as a request to quit.
        if isinstance(value, KeyboardInterrupt):
            QApplication.instance().quit()
            return
```
but while it exits cleanly sometimes, sometimes it bus errors or seg faults.

**Comment by tacaswell** [comment]
xref same discussion at Matplotlib https://github.com/matplotlib/matplotlib/pull/13306 and a (maybe janky) implementation from my DAMA work https://github.com/bluesky/bluesky/blob/cde1d87294a41d48d0a92d6cae0ce9c508e20388/bluesky/utils.py#L1384-L1495
